src/storage.rs

```diff
@@ -13,19 +14,41 @@ static DEFAULT_PATH: &'static str = "~/Library/Emerald";
 static DEFAULT_PATH: &'static str = "%APPDATA%\\.emerald";
 
 /// Base dir for internal data, all chain-related should be store in subdirectories
-#[derive(Debug, Clone)]
-pub struct Storages<'a> {
+#[derive(Debug, Clone)] 
+pub struct Storages {
     /// base dir
-    base_dir: &'a Path,
+    base_dir: PathBuf,
+}
+
+#[cfg(all(unix, not(target_os = "macos"), not(target_os = "ios"), not(target_os = "android")))]
+pub fn default_path() -> PathBuf {
+    let mut config_dir = env::home_dir().unwrap();
+    config_dir.push(".emerald");
+    config_dir
+}
+
+#[cfg(target_os = "macos")]
+pub fn default_path() -> PathBuf {
+    let mut config_dir = env::home_dir().unwrap();
+    config_dir.push("Library");
+    config_dir.push("Emerald");
+    config_dir
 }
 
-impl<'a> Storages<'a> {
-    /// Create storage using user directory if specified,
+#[cfg(target_os = "windows")]
+pub fn default_path() -> PathBuf {
+    let mut config_dir = env::var("APPDATA").unwrap();
```

Comment:
Please don't use `unwrap` where it's 100% not guaranteed, change to `.expect("Expect 'APPDATA' environment variable")`.